packages/platform/atoms/hooks/connect/useCheck.ts

```diff
@@ -1,20 +1,30 @@
 import { useQuery, useQueryClient } from "@tanstack/react-query";
 
 import type { CALENDARS } from "@calcom/platform-constants";
-import { SUCCESS_STATUS, ERROR_STATUS } from "@calcom/platform-constants";
-import type { ApiResponse, ApiErrorResponse } from "@calcom/platform-types";
+import { ERROR_STATUS, SUCCESS_STATUS } from "@calcom/platform-constants";
+import type { ApiErrorResponse, ApiResponse } from "@calcom/platform-types";
 
 import http from "../../lib/http";
 import { useAtomsContext } from "../useAtomsContext";
 
 export interface UseCheckProps {
   onCheckError?: OnCheckErrorType;
   calendar: (typeof CALENDARS)[number];
+  config?: Partial<{
```

Comment:
I think we can reduce nesting and simply have `initialData` prop for `UseCheckProps` instead of `config` - we currently only have 1 config property `initialData`, so I don't see a reason to nest it in config - if we will have more use query related props, we can then think of config object.

Reply:
Ah makes sense. My reasoning was to allow this to extend in future with additional react query props like you mentioned without havving to break the code but agree since in this case we only need intialData, we can reduce the nesting